### PR TITLE
*: parse and ignore lock option in alter table statement

### PR DESCRIPTION
**Pull request by shenli**

### CODE REVIEW

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -704,12 +704,21 @@ func handleTableOptions(options []*ast.TableOption, tbInfo *model.TableInfo) {
...
+	// Only handle valid specs, AlterTableLock is ignored.
+	validSpecs := make([]*ast.AlterTableSpec, 0, len(specs))
+	for _, spec := range specs {
+		if spec.Tp == ast.AlterTableLock {
+			continue
+		}
+		validSpecs = append(validSpecs, spec)
+	}
+
+	if len(validSpecs) != 1 {
```
Comment (zimulala) [inline ddl/ddl_api.go]: How about length is 0？

Comment (shenli) [inline ddl/ddl_api.go]: The same with the old code.

File: ddl/ddl_api.go [hunk 2]
```diff
@@ -704,12 +704,21 @@ func handleTableOptions(options []*ast.TableOption, tbInfo *model.TableInfo) {
 }
 
 func (d *ddl) AlterTable(ctx context.Context, ident ast.Ident, specs []*ast.AlterTableSpec) (err error) {
-	// Now we only allow one schema changing at the same time.
-	if len(specs) != 1 {
+	// Only handle valid specs, AlterTableLock is ignored.
+	validSpecs := make([]*ast.AlterTableSpec, 0, len(specs))
+	for _, spec := range specs {
```
Comment (coocood) [inline ddl/ddl_api.go]: Why not just add a case `ast.AlterTableLock` in the switch below?

Comment (coocood) [inline ddl/ddl_api.go]: never mind, we need to check spec is one.

Comment (shenli) [inline ddl/ddl_api.go]: Then the statement will meet error at Line 708.

**Comment by shenli** [comment]
@coocood @zimulala PTAL

**Comment by zimulala** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**